### PR TITLE
Change Rake tasks

### PR DESCRIPTION
**Pull request by jmauro**
STATE:
Currently, the kitchen-test are not run in parallel

FIX:
- Split the rake file into differents pieces
- Add concurrency for kitchen test

Change-Id: Ic6cf762f4bda29df412a5e670ad4406872fe47bf

### CODE REVIEW

**Comment by jmauro** [comment]
Hello @therobot,

All suite are active and the entire build takes 18 mins, isn't that create?

**Comment by jmauro** [comment]
'great' better?

**Comment by jmauro** [comment]
Yes I tried what @someara told me (Bumping to 2.2.3)

**Comment by jmauro** [comment]
Reverting back to an old version of kitchen-dokken

**Comment by jmauro** [comment]
@therobot test seems to pass ;)

**Comment by jmauro** [comment]
Finally, test seem stable. Launch them twice to make sure there were no inspec bug.

@therobot any chance this will be merged soon?

**Comment by jmauro** [comment]
Hello,

Finally, I used the `matrix` feature to split test by function and also add a concurrency setting.

File: rakelib/35-rspec.rake [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  # Rspec and ChefSpec
+  desc 'Run ChefSpec examples'
+  RSpec::Core::RakeTask.new(:chefspec) do |t|
+    t.rspec_opts = %w(
+      --color
+      --format progress
+    ).join(' ')
+  end
+end
+task unit: %w( unit:chefspec )
```
Comment (onlyhavecans) [inline rakelib/35-rspec.rake]: Why duplicate this code both here and in 99-rake-tasks?

**Comment by jmauro** [comment]
@onlyhavecans splitting the rakefile is just to break thing down, better isolation and avoiding reading a huge rakefile for just changing for instance a rubucop rules, but maybe you are right this could be overkill.

let me know if you want me to merge them back, I just find that easier to read.

Regards,
JM

File: rakelib/35-rspec.rake (hunk 1 above)
Comment (jmauro) [inline rakelib/35-rspec.rake]: Fixed

**Comment by onlyhavecans** [comment]
I'm Ok with it split up. My only "change request" was the duplicated code.

**Comment by jmauro** [comment]
The fix was provided :D.

File: rakelib/30-foodcritic.rake [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  desc 'Run Chef style checks'
+  FoodCritic::Rake::LintTask.new(:foodcritic) do |t|
+    t.options = {
+      fail_tags: ['any']
+    }
+  end
+end
+
+task style: %w( style:rubocop style:foodcritic )
```
Comment (onlyhavecans) [inline rakelib/30-foodcritic.rake]: Sorry, this is also duplicate code

Comment (jmauro) [inline rakelib/30-foodcritic.rake]: good catch ;)

Comment (jmauro) [inline rakelib/30-foodcritic.rake]: Fix on its way.

**Pull request merged**